server/game/cards/10-SoD/TyeneSand.js

```diff
@@ -0,0 +1,22 @@
+const DrawCard = require('../../drawcard.js');
+
+class TyeneSand extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                afterChallenge: event => event.challenge.loser === this.controller && event.challenge.defendingPlayer === this.controller
+            },
+            location: 'hand',
+            condition: () => this.controller.canPutIntoPlay(this),
```

Comment:
`condition` doesn't work with reactions / interrupts, you'll need to check this in the `when` handler.

I really should add some checks when creating the ability to help catch things like this.